src/index.js

```diff
@@ -32,3 +32,39 @@ export function parseExpression(input, options) {
 
 
 export { tokTypes };
+
+function getParser(options, input) {
+  const cls = options && options.plugins ? getParserClass(options.plugins) : Parser;
+  return new cls(options, input);
+}
+
+const parserClassCache = {};
+
+/** Get a Parser class with plugins applied. */
+function getParserClass(pluginsFromOptions) {
+  // Filter out just the plugins that have an actual mixin associated with them.
+  let pluginList = pluginsFromOptions.filter((p) => p === "estree" || p === "flow" || p === "jsx");
+
+  if (pluginList.indexOf("flow") >= 0) {
+    // ensure flow plugin loads last
+    pluginList = pluginList.filter((plugin) => plugin !== "flow");
+    pluginList.push("flow");
+  }
+
+  if (pluginList.indexOf("estree") >= 0) {
+    // ensure estree plugin loads first
+    pluginList = pluginList.filter((plugin) => plugin !== "estree");
+    pluginList.unshift("estree");
+  }
+
+  const key = pluginList.join("/");
+  let cls = parserClassCache[key];
+  if (!cls) {
+    cls = Parser;
+    for (const plugin of pluginList) {
+      cls = plugins[plugin](cls);
+    }
+    parserClassCache[key] = cls;
+  }
+  return  cls;
```

Comment:
`return  cls;` guess we didn't have a rule for this, I guess we should land that prettier pr later